CONTRIBUTING.md

```diff
@@ -28,65 +28,118 @@ Here are questions you can answer before you file a bug to make sure you're not
-    breadcrumbs for the next person debugging your code (who might be you).
-  * Using the words `Fixes`, `Fixed`, `Closes`, `Closed`, or `Close` followed by
-    `#<issuenumber>`, will auto close an issue after the PR is merged.
-* Be a child commit of a reasonably recent commit in the main branch
-* Pass all unit tests
-* Ensure that the title and description are fully up to date before merging
-  * The title and description feed the final git commit message, and we want to
-    ensure high quality commit messages in the repository history.
-* Include adequate tests
-  * At least one test should fail in the absence of your non-test code changes. If your PR does not match this criteria, please specify why
-  * Tests should include reasonable permutations of the target fix/change
-  * Include baseline changes with your change as needed
+#### Titles
+
+The title should focus on what the change intends to do rather than how it was done.
+The description can and should explain how it was done if not obvious.
+
+Titles under 76 characters print nicely in unix terminals under `git log`.
+This is not a hard requirement, but is good guidance.
+
+Tags in titles allow for speedy categorization
+Title tags  are generally one word or acronym enclosed in square brackets.
+Limiting to one or two tags is ideal to keep titles short.
+Some examples of common tags are:
+
+- `[NFC]` - No Functional Change
+  - `[RFC]` - Request For Comments (often used for drafts to get feedback)
+  - `[Doc]` - Documentation change
+  - `[SPIRV]` - Changes related to SPIR-V
+  - `[HLSL2021]` - Changes related to HLSL 2021 features
+  - Other tags in use: `[Linux]`, `[mac]`, `[Win]`, `[PIX]`, etc...
+
+Tags aren't formalized or any specific limited set. If you're unsure of
+  a reasonable tag to use, just don't use any. If you want to invent a new
+  tag, go for it! These are to help categorize changes at a glance.
+
+#### Descriptions
+
+The PR description should include a more detailed description of the change,
+an explanation for the motivation of the change, and links to any relevant Issues.
+This does not need to be a dissertation, but should leave
+breadcrumbs for the next person debugging your code (who might be you).
+
+Using the words `Fixes`, `Fixed`, `Closes`, `Closed`, or `Close` followed by
+  `#<issuenumber>`, will auto close an issue after the PR is merged.
+
+#### Release Notes
+
+Significant changes may require release notes that highlight important
+compiler behavior changes for each named release.
+These include changes that are:
+
+- Visible to the users
+- Significant changes to compiler behavior
+  - New features: Language, Hardware support, compiler options
+  - Important bug fixes
+  - Changes in default behavior
+
+When such a change is made, the release note should be included as part of that change.
+This is done in the docs/ReleaseNotes.md file.
+
+If the change is meant for a named release,
```

Comment:
can we wrap lines consistently please?  I find this confusing to read - it looks like a poem.  Maybe I've missed a literary reference?

Reply:
I write MD files this way in order to make follow-up changes easier to review since trying to review a long line of MD for changes somewhere therein is painful. 

The MD processor eliminates all single line breaks, so you can read it cleanly if you use "rich diff":
![image](https://github.com/user-attachments/assets/3fd1f677-1358-406a-9fc7-058bf7e5f883)

Reply:
My feedback here is that I find this style unusual and it makes reading it harder.

Reply:
I've relaxed my line length threshold and merged most of these lines while making those left look more like continuation lines.